### PR TITLE
Add Msaboot tool

### PR DESCRIPTION
**Pull request by matthew-fogel**
New tool for bootstrapping multiple sequence alignment files.

### CODE REVIEW

File: tools/msaboot/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+categories:
+- Fasta Manipulation
+description: A multiple sequences alignment bootstrapping tool.
+homepage_url: https://github.com/phac-nml/msaboot
+long_description: This tool can be used to generate bootstrapped replicates of multiple sequence alignments. The bootstrapping is done on the columns of the alignment.
+name: msaboot
+owner: phac-nml
```
Comment (bgruening) [inline tools/msaboot/.shed.yml]: This tool is not yet in the toolshed, so we could make the owner IUC if you agree.

File: tools/msaboot/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+remote_repository_url: https://github.com/phac-nml/msaboot
```
Comment (bgruening) [inline tools/msaboot/.shed.yml]: https://github.com/galaxyproject/tools-iuc/tree/master/tools/msaboot

File: tools/msaboot/msaboot.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0"?>
+<tool id="msaboot" name="MSABOOT" version="@VERSION@">
+    <description>Output PHYLIP file with bootstrapped multiple sequence alignment data.</description>
```
Comment (bgruening) [inline tools/msaboot/msaboot.xml]: the dot can be removed.

File: tools/msaboot/msaboot.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        <token name="@VERSION@">0.1.0</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">msaboot</requirement>
+    </requirements>
+    <version_command>msaboot</version_command>
+    <command detect_errors="aggressive">
+        <![CDATA[
+	    msaboot
+      -i $input -n $number -o $output > $log 2>&1
```
Comment (bgruening) [inline tools/msaboot/msaboot.xml]: * input and output needs to be single quoted
* the indentation could be adjusted

File: tools/msaboot/msaboot.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        ]]>
+    </command>
+    <inputs>
+        <param name="input" format="fasta" type="data" label="FASTA input file" />
+        <param name="number" type="integer" value="1" label="Specify the number of bootstrapping replicates to generate." />
```
Comment (bgruening) [inline tools/msaboot/msaboot.xml]: do you think a min="1" would make sense?

File: tools/msaboot/msaboot.xml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    </inputs>
+    <outputs>
+        <data name="output" format="phylip" label="${tool.name} on ${on_string}:out.phylip" />
+        <data name="log" format="txt" hidden="true" label="${tool.name} on ${on_string}:log_msabootpl_run.txt"/>
```
Comment (bgruening) [inline tools/msaboot/msaboot.xml]: should the log file be an optional output? In this case you could add a boolean parameter asking for this log file.

File: tools/msaboot/msaboot.xml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+You can unhide the hidden output by clicking on the cog wheel next to the History panel and select "Unhide Hidden Datasets".
+
+(A) The bootstrapped replicates in Phylip format.
+
+(B) The hidden log file, containing information about the msaboot run to bootstrap the multiple sequence alignment data.
+
+]]>
+    </help>
+    <citations>
```
Comment (bgruening) [inline tools/msaboot/msaboot.xml]: You could add here some bibtex to reference your github repo, like here: https://github.com/galaxyproject/tools-iuc/blob/eea727c3bdfe36d9d16036d5ab79fb8b27c4e82e/tools/umi_tools/macros.xml#L6

File: tools/msaboot/msaboot.xml (hunk 6 above)
Comment (matthew-fogel) [reply tools/msaboot/msaboot.xml]: I will make this log file an optional output, thank you for your advice!

File: tools/msaboot/.shed.yml (hunk 1 above)
Comment (matthew-fogel) [reply tools/msaboot/.shed.yml]: Yes, I will change it accordingly.

File: tools/msaboot/msaboot.xml (hunk 3 above)
Comment (matthew-fogel) [reply tools/msaboot/msaboot.xml]: Thank you, I have removed it.

File: tools/msaboot/msaboot.xml (hunk 5 above)
Comment (matthew-fogel) [reply tools/msaboot/msaboot.xml]: Yes.

File: tools/msaboot/.shed.yml (hunk 2 above)
Comment (matthew-fogel) [reply tools/msaboot/.shed.yml]: Thank you again!

**Comment by matthew-fogel** [comment]
Thank you for your review(s)! I added a sample log file for the test case and other than that implemented the changes that you suggested.

**Comment by bgruening** [comment]
@matthew-fogel thanks a lot!

**Pull request merged**